### PR TITLE
Add support for data representation

### PR DESCRIPTION
**Pull request by MiguelCompany**
This fixes the failures mentioned [here](https://github.com/ros2/ros2/pull/1560#issue-2286372464) by implementing the new `is_plain` overload of Fast DDS type support interface.

It additionally sets the data representation policy of created DataReader and DataWriter entities to XCDR1, since it is what the type support is using.

### CODE REVIEW

**Comment by MiguelCompany** [comment]
@clalancette @ahcorde @MichaelOrlov FYI

This makes ROS 2 CI pass (I checked locally on both Windows and Ubuntu) with the current head of Fast DDS 2.14.x, making it possible to revert https://github.com/ros2/ros2/pull/1560

**Review by ahcorde — COMMENTED** [review]
can we add a test ?

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21030/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15332/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21724/)

**Comment by clalancette** [comment]
Because of where we are in the release process for Jazzy, I don't think we should put this in at this time.

What I think we should do right now is to get this PR ready, but not merge it yet.  That is, we keep both Rolling and Jazzy pinned to the known working commit for Fast-DDS for now.  Once Fast-DDS has made a tag available with the breaking change in it, we merge this into Rolling and then backport to Jazzy (that will be for Jazzy patch release 1).

@marcoag FYI.

**Comment by EduPonz** [comment]
> What I think we should do right now is to get this PR ready, but not merge it yet. That is, we keep both Rolling and Jazzy pinned to the known working commit for Fast-DDS for now. Once Fast-DDS has made a tag available with the breaking change in it, we merge this into Rolling and then backport to Jazzy (that will be for Jazzy patch release 1).

I think that is a sensible thing to do. Just by chance I'm working on releasing Fast DDS v2.14.1 between today and tomorrow. I'll keep you posted.

**Comment by EduPonz** [comment]
@clalancette @ahcorde In case you want to move this fordward, we just released Fast DDS v2.14.1:

* Rolling PR: ros/rosdistro#41140
* Jazzy PR: ros/rosdistro#41141

**Comment by clalancette** [comment]
> * Rolling PR:

Oh, I am going to close both of those.  They can't possibly compile right now, because we don't have this PR in.

**Comment by MiguelCompany** [comment]
> > * Rolling PR:
> 
> Oh, I am going to close both of those. They can't possibly compile right now, because we don't have this PR in.

They will compile, but tests for loaned messages will fail until this one is merged.

**Comment by MiguelCompany** [comment]
@clalancette Can we move forward with this?

**Review by fujitatomoya — APPROVED** [review]

**Comment by fujitatomoya** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21165/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15470/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21889/)

**Pull request merged**
